### PR TITLE
added route for firmware install

### PR DESCRIPTION
**Pull request by HariHive**
- [ ] Add a link to the ticket in the PR title or add a description of work in the PR title
- [ ] dashcam firmware was loaded on a device and successfully runs
- [ ] semantic version was incremented inside `src/config/index.ts`

### CODE REVIEW

**Comment by HariHive** [comment]
Tested with HDC and HDCS using postman

File: src/routes/firmware.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    try {
+        if (CAMERA_TYPE === CameraType.Hdc) {
+            try {
+                execSync(`test -f ${HDC_ROOT + firmwareFile}`, {
+                    encoding: 'utf-8',
+                });
+            } catch (error: unknown) {
+                console.log("Rauc file is not present")
+            }
+            runSpawn(`rauc install ${HDC_ROOT + firmwareFile} && reboot`)
```
Comment (punov) [inline src/routes/firmware.ts:52]: @HariHive @harroyoc 
Have we had a plan to leave the phone a chance to wait for installation to finish and sent "reboot" command?
Or you guys want the dashcam to fully own it?

There's a point of letting the dashcam restart itself, but I'm worried if the phone will be able to properly handle and receive the updated state in this case

File: src/routes/firmware.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                console.log("Mender file is not present")
+            }
+            try {
+                execSync(`test -f ${FIP_PATH}  && rm ${FIP_PATH}`, {
+                    encoding: 'utf-8',
+                });
+            } catch (error: unknown) {
+                console.log("Fip file is not present")
+            }
+            runSpawn(`tar -xzf /data/${firmwareFile} -C /data && os-update --install ${MENDER_PATH} && movisoc-fwu -a ${FIP_PATH} && reboot`)
```
Comment (punov) [inline src/routes/firmware.ts:70]: @HariHive 
What we should check here for sure is that if mender for some reason errors out but doesn't break the command chain, we don't want movisoc-fwu to be called.
So we should be 100% sure that failing mender (os-update) will break the command chain for sure

File: src/routes/firmware.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            res.json({ output: 'received install command' });
+        }
+    } catch (error: unknown) {
+        res.json({ error });
+    }
+});
+
+router.get('/progress', async (req, res) => {
+    res.json({ message });
```
Comment (punov) [inline src/routes/firmware.ts:79]: @HariHive I deally we give the percentage number back, not just the last message.
So adding the logic to unparse the progress will be good.

progress = 0
....
  on.data -> reads message, parses percentage out of it (when possible), assigns to progress (if successfully parsed)
....
/progress route:
returns progress value

File: src/routes/firmware.ts (hunk 1 above)
Comment (HariHive) [reply src/routes/firmware.ts:52]: Discussed offline, we want app to handle reboot

File: src/routes/firmware.ts (hunk 2 above)
Comment (HariHive) [reply src/routes/firmware.ts:70]: could not check if os-update command breaks, but test with tar command, if tar fails nothing executes:
image

**Comment by HariHive** [comment]
closing as duplicate

**Pull request closed without merging**